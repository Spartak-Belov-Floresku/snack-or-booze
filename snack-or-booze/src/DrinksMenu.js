import { Link } from "react-router-dom";
import "./FoodMenu.css";
import {
  Card,
  CardBody,
  CardTitle,
  ListGroup,
  ListGroupItem
} from "reactstrap";


/*
  component will create drink menu and displayed on the UI
*/
function DrinksMenu({ drinks }) {
  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            Drink Menu
          </CardTitle>
          <ListGroup>
            {drinks.map(drink => (
              <Link to={`/drinks/${drink.id}`} key={drink.id}>
                <ListGroupItem>{drink.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
          <Link to="/"><ListGroupItem>Home page</ListGroupItem></Link>
        </CardBody>
      </Card>
    </section>
  );
}

export default DrinksMenu;