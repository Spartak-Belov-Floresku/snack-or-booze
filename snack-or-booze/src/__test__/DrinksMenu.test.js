import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import SnackOrBoozeApi from "../Api";
import DrinksMenu from '../DrinksMenu';

let drinks;

/**befor ech tests get data from database */
beforeEach( async () => {

  function getDrinks(){
    return SnackOrBoozeApi.getDrinks();
  }    
  
  drinks = await getDrinks();

});


/** render DrinksMenu component */
test('renders learn react link', () => {

    render(
        <BrowserRouter>
            <DrinksMenu drinks={drinks} />
        </BrowserRouter>
    );
  
});

/** Matching snapshot of DrinksMenu components */
it("matches snapshot Home component", () => {  
      
    const {asFragment} = render(
        <BrowserRouter>
            <DrinksMenu drinks={drinks} />
        </BrowserRouter>
    );
    
    expect(asFragment()).toMatchSnapshot();
});

/** check if the menu exists on the DrinkMenu page */
it("menu on the home page", () => {

    const { queryByText } = render(
        <BrowserRouter>
            <DrinksMenu drinks={drinks}/>
        </BrowserRouter>
    );
    expect(queryByText("Drink Menu")).toBeInTheDocument();

});

/** checking if links to snack exists in DrinksMenu bar*/
it("check link", () => {

    const { getByText } = render(
        <BrowserRouter>
            <DrinksMenu drinks={drinks} />
        </BrowserRouter>
    );

    expect(screen.getByText('Gin and Tonic').closest('a')).toHaveAttribute('href', '/drinks/gin-and-tonic');

});
