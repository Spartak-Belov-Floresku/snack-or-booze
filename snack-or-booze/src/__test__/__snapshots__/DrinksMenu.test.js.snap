// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot Home component 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Drink Menu
        </h5>
        <ul
          class="list-group"
        >
          <a
            href="/drinks/martini"
          >
            <li
              class="list-group-item"
            >
              Martini
            </li>
          </a>
          <a
            href="/drinks/negroni"
          >
            <li
              class="list-group-item"
            >
              Negroni
            </li>
          </a>
          <a
            href="/drinks/gin-and-tonic"
          >
            <li
              class="list-group-item"
            >
              Gin and Tonic
            </li>
          </a>
          <a
            href="/drinks/diet-coke"
          >
            <li
              class="list-group-item"
            >
              Diet Coke
            </li>
          </a>
        </ul>
        <a
          href="/"
        >
          <li
            class="list-group-item"
          >
            Home page
          </li>
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
